commons-numbers-fraction/src/main/java/org/apache/commons/numbers/fraction/Fraction.java

```diff
@@ -775,19 +778,45 @@ public Fraction divide(Fraction value) {
      */
     @Override
     public Fraction pow(final int exponent) {
+        if (exponent == 1) {
+            return this;
+        }
         if (exponent == 0) {
             return ONE;
         }
-        if (isZero()) {
-            return ZERO;
+        if (exponent == -1) {
+            return this.reciprocal();
         }
-
-        if (exponent < 0) {
-            return new Fraction(ArithmeticUtils.pow(denominator, -exponent),
-                                ArithmeticUtils.pow(numerator,   -exponent));
+        if (this.denominator == 0) {
```

Comment:
The private constructor should make it impossible to have a denominator of zero. See if you can write a test that hits this code execution path.

This functionality is better served by the current logic that checks if (isZero()).

Reply:
@aherbert 
Yes, you are correct.